lib/AST/GraphOperationInfo.cpp

```diff
@@ -74,35 +74,16 @@ void GraphOperationInfo::assertWithDump(bool cond,
 
 /// Decode the name of a graph_op into its TensorFlow op name and a list of
 /// information about the operands.
-StringRef
-GraphOperationInfo::decodeName(SmallVectorImpl<InputMarker> &inputInfo) {
+StringRef GraphOperationInfo::decodeName(
+    SmallVectorImpl<OperandMarker> &operandMarkers) const {
   auto name = inst->getName().str();
   auto pos = name.find(',');
   auto opName = name.substr(0, pos);
 
   while (pos != StringRef::npos) {
-    name = name.drop_front(pos + 1);
-    pos = name.find(',');
-    auto letter = name.substr(0, pos);
-    assertWithDump(letter.size() == 1, "malformed graph_op instruction");
-    InputMarker kind;
-    switch (letter[0]) {
-    case 's':
-      kind = InputMarker::IM_Scalar;
-      break;
-    case 'i':
-      kind = InputMarker::IM_Normal;
-      break;
-    case 'L':
-      kind = InputMarker::IM_InputList;
-      break;
-    case 'e':
-      kind = InputMarker::IM_InputListElt;
-      break;
-    default:
-      assertWithDump(false, "malformed graph_op instruction");
-    }
-    inputInfo.push_back(kind);
+    name = name.drop_front(pos);
+    pos = name.find(',', 1);
+    operandMarkers.push_back(OperandMarker(name.substr(0, pos), *this));
```

Comment:
Can we avoid passing *this if possible? It seems to be only used for assertWithDump().

Reply:
yes! I found a way with `PrettyStackTraceSILNode` to make it dump out the bad instruction when there is an assertion failure, without needing to call assertWithDump().